code/code/spec/spec_objs_thief_quest.cc

```diff
@@ -0,0 +1,62 @@
+#include <stdio.h>
+
+#include "being.h"
+#include "comm.h"
+#include "extern.h"
+#include "obj_base_weapon.h"
+
+int thiefQuestWeapon(TBeing *victim, cmdTypeT command, const char *arg, TObj *object, TObj *) {
+  wearSlotT limb = WEAR_NOWHERE;
+
+  if (command != CMD_STAB && command != CMD_BACKSTAB && command != CMD_SLIT) return false;
+  if (command == CMD_STAB) limb = static_cast<wearSlotT>(reinterpret_cast<uintptr_t>(arg));
```

Comment:
So when we get to this conversion we know that this is coming in with the CMD_STAB and you've coded CMD_STAB to pass in the limb. There should be no way that something else could get here that's not what you expect. 

So it looks kinda scary but I think it should be safe.

Let's see if @cizra has an opinion on this bit.

Reply:
Right - not sure if adding the extra code to convert cleanly between enum/string or doing these ugly casts is more preferable here. The code seems to work fine either way though, based on my fairly extensive in-game testing/debugging of it.

Until we tackle the job of rewriting the checkSpec function to accept typed unions or variants or whatever as parameters, and therefore also update the function signatures of every single spec proc that gets called by checkSpec, we'll have to find less favorable workarounds like this, I think. :\

Reply:
I don't know of a better way of doing it, except for https://en.cppreference.com/w/cpp/utility/variant - and if we're not ready to switch to varianst, well, we gotta reinterpret_cast them :(